_posts/2021-01-15-docviewer-updates.md

```diff
@@ -0,0 +1,100 @@
+
+Not only the code has become simpler but also features. For example, rotating pages is only a form of visual representation, so it is unnecessary to store this information, it can be reproduced anytime. SearchBar widget become redundant too, because all browsers can search on page, so we removed this from selectable widgets. ✂️
+
+We've put a great emphasis on improving performance as well. Sometimes we can achieve significant results by making small changes, such as taking advantage of the opportunities provided by our framework: replacing useEffect with [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo). 🚀
+
+Which is also important during these kind of full review is finding and fixing bugs in the software. Of course, during the re-rolling of this function, we also encountered many cases where a different solution had to be applied than we originally invented. 🐛
+
+The point is that we tried to look at the possibilities from the user's point of view and we were trying to make this feature user friendly.👭
+
+# Style 🎨
+
+We are also spreading another new option to users, which is overwriting styles from a higher level. It is very important that the user who uses our service can shape it for her/his own needs. So far there have been relatively few opportunities for it, but we are changing that now:
+
+You can define you style on your own and you can pass it to the component itself, like:
+
+```javascript
+const useDisabledStyle = makeStyles(() =>
+  createStyles({
+    iconButton: {
+      "&:disabled": {
+        opacity: 0.26,
+      },
+    },
+  })
+);
+
+const disabledStyle = useDisabledStyle();
+```
+
+If you use typescript you can also see the possible classes what you can overwrite:
+
+<p align="center">
+<img src="/img/posts/docviewer/overwrite_styles.png">
+</p>
+
+Our designer is constantly coming up with new solutions, the style is constantly changing, we can never consider it ready. 🎡
+
+# New features 🎉
+
+For those who have been using sensenet for a long time, it may not be new, but if you have just joined, you may come across the keywords redaction, highlight and annotation for the first time.
+
+**Redaction** 🔳
+
+Sometimes when you are working with sensitive information 🗝️ in documents you don’t want to share them with everyone, just who has the appropriate privileges. But what if you would like to share the other parts of the document? This is the case when you need to apply redaction.
+With this feature you can hide parts of your doc with a black rectangle. If another user would like to see it, it is only possible if she/he has the permission for it, otherwise the page will be full with black squares. 🙈
```

Comment:
```suggestion
With this feature you can hide parts of your doc with a black rectangles. Opening a document without redaction is only possible if the user has permission for it, otherwise the page will be full with black squares. 🙈
```